From b8974cc7a4c2e5d8cfdce3a6e06a72a540fff373 Mon Sep 17 00:00:00 2001
From: David Sommerseth <davids@openvpn.net>
Date: Wed, 22 Jan 2025 23:36:55 +0100
Subject: [PATCH] client: Handle exceptions in ~BackendStarterSrv

The log service Detach call may fail in some cases, this should not
crash the openvpn3-service-backendstart program.

Signed-off-by: David Sommerseth <davids@openvpn.net>
---
 src/client/openvpn3-service-backendstart.cpp | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/client/openvpn3-service-backendstart.cpp b/src/client/openvpn3-service-backendstart.cpp
index 2bf4333..f0a03cc 100644
--- a/src/client/openvpn3-service-backendstart.cpp
+++ b/src/client/openvpn3-service-backendstart.cpp
@@ -339,7 +339,15 @@ class BackendStarterSrv : public DBus::Service
 
     ~BackendStarterSrv()
     {
-        logsrvprx->Detach(Constants::GenInterface("backends"));
+        try
+        {
+            logsrvprx->Detach(Constants::GenInterface("backends"));
+        }
+        catch (const DBus::Exception &excp)
+        {
+            std::cerr << "[EXCEPTION] ~BackendStarterSrv(): "
+                      << excp.what() << std::endl;
+        }
     }
 
 
